src/P2P/Node/PeerDB.hs

```diff
@@ -175,8 +180,21 @@ instance Arbitrary PeerEntry where
 -- -------------------------------------------------------------------------- --
 -- Peer Entry Set
 
+pdNonce :: Int
+pdNonce = unsafePerformIO randomIO
```

Comment:
Consider generating this in `newEmptyPeerDb`

Reply:
This is safe because all `randomIO` needs is an incrementing state token to be sufficiently random. For ergonomics' sake, I agree with @larskuhtz here. Less to think about/deal with and it does roughly the same thing.